src/components/annotations/draw.js

```diff
@@ -78,8 +78,12 @@ function drawOne(gd, index) {
 // the plot.
 // axDomainRef: if true and axa defined, draws relative to axis domain,
 // otherwise draws relative to data (if axa defined) or paper (if not).
-function shiftPosition(axa, optAx, dAx, gsDim, vertical, axRef) {
+function shiftPosition(axa, options, dAx, axLetter, gs) {
```

Comment:
Nice simplifications! Thanks.
In plotly.js we often pass `options` as the last argument of the function. And we could read `axLetter` from the first argument. So perhaps the function should look like:
```js
function shiftPosition(axa, dAx, gs, options) {
  var axLetter = axa._id.charAt(0);
```

Reply:
sometimes `axLetter` is two characters, for example for annotation arrows it is `ay` or `ax`.

Reply:
Good call. So let's keep `axLetter` in the arguments and only move the options to the end. Thanks.

Reply:
fixed by 759d37bf9e1eac56cef5d1e4df44e899e20ffadc